providers/terraform-provider-csbmssqldbrunfailover/connector/connector.go

```diff
@@ -0,0 +1,79 @@
+// Package connector manages the failover state in MSSQL DB
+package connector
+
+import (
+	"context"
+	"fmt"
+	"os"
+
+	"github.com/Azure/azure-sdk-for-go/sdk/azidentity"
+	"github.com/Azure/azure-sdk-for-go/sdk/resourcemanager/sql/armsql"
+)
+
+type Connector struct {
+	azureTenantID, azureClientID, azureClientSecret, azureSubscriptionID string
+}
+
+func NewConnector(azureTenantID, azureClientID, azureClientSecret, azureSubscriptionID string) *Connector {
+	return &Connector{
+		azureTenantID:       azureTenantID,
+		azureClientID:       azureClientID,
+		azureClientSecret:   azureClientSecret,
+		azureSubscriptionID: azureSubscriptionID,
+	}
+}
+
+func (c *Connector) CreateRunFailover(ctx context.Context, resourceGroup, serverName, failoverGroup string) error {
```

Comment:
This function is used for both create and delete. Is there a better name for it?

Reply:
Good catch, @blgm . What do you think about `SwitchFailover`?